### PR TITLE
Added option to enable multi threaded compilation on Windows

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
* Moved project on top to resolve the generator as early as possible
* Added information about the flag to the README

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,13 +17,20 @@
...
+project(olp-cpp-sdk VERSION 0.6.0)
+
 # Options
 option(EDGE_SDK_ENABLE_TESTING "Flag to enable/disable building unit and integration tests" ON)
 option(EDGE_SDK_BUILD_DOC "Build SDK documentation" OFF)
 option(EDGE_SDK_NO_EXCEPTION "Turn off EXCEPTION" OFF)
 option(EDGE_SDK_BUILD_EXTERNAL_DEPS "Download and build external dependencies" ON)
 option(EDGE_SDK_BUILD_EXAMPLES "Enable examples targets" OFF)
 
+if (MSVC)
```
Comment (andescu) [inline CMakeLists.txt]: Do we need to guard config variables by platform? They should have no effect either way.

File: CMakeLists.txt [hunk 2]
```diff
@@ -33,12 +40,14 @@ if (EDGE_SDK_NO_EXCEPTION)
     set(CMAKE_C_FLAGS "-fno-exceptions ${CMAKE_C_FLAGS}")
 endif()
 
+if (MSVC AND EDGE_SDK_PARALLEL_BUILD_ENABLE)
+    add_compile_options(/MP)
+    message(STATUS "Using parallel compiling (/MP)")
```
Comment (andescu) [inline CMakeLists.txt]:
```suggestion
    message(STATUS "MSVC  - Using parallel compilation (/MP)")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -33,12 +40,14 @@ if (EDGE_SDK_NO_EXCEPTION)
...
+endif ()
```
Comment (borzun) [inline CMakeLists.txt]: I saw you used `endif(MSVC)` in previous if-clause. Shouldn't it we use it here as well?
```suggestion
endif (MSVC AND EDGE_SDK_PARALLEL_BUILD_ENABLE)
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -17,13 +17,20 @@
...
+
 # Options
 option(EDGE_SDK_ENABLE_TESTING "Flag to enable/disable building unit and integration tests" ON)
 option(EDGE_SDK_BUILD_DOC "Build SDK documentation" OFF)
 option(EDGE_SDK_NO_EXCEPTION "Turn off EXCEPTION" OFF)
 option(EDGE_SDK_BUILD_EXTERNAL_DEPS "Download and build external dependencies" ON)
 option(EDGE_SDK_BUILD_EXAMPLES "Enable examples targets" OFF)
 
+if (MSVC)
+option(EDGE_SDK_PARALLEL_BUILD_ENABLE "Enable parallel build on MSVC" ON)
```
Comment (borzun) [inline CMakeLists.txt]: Ident is missing. 
```suggestion
    option(EDGE_SDK_PARALLEL_BUILD_ENABLE "Enable parallel build on MSVC" ON)
```

Comment (borzun) [inline CMakeLists.txt]: Better name it platform dependent, like EDGE_SDK_MSVC_PARALLEL_BUILD_ENABLE:
```suggestion
option(EDGE_SDK_MSVC_PARALLEL_BUILD_ENABLE "Enable parallel build on MSVC" ON)
```

Btw, do we really need this flag? Can't the user just pass /MP flag to CMAKE_CXX_FLAGS?

File: CMakeLists.txt (hunk 2 above)
Comment (mykhailo-kuchma) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 4 above)
Comment (mykhailo-kuchma) [reply CMakeLists.txt]: Done

Comment (mykhailo-kuchma) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 3 above)
Comment (mykhailo-kuchma) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 1 above)
Comment (mykhailo-kuchma) [reply CMakeLists.txt]: Removed the guard

**Pull request merged**